spec/spec_helper.rb

```diff
@@ -81,7 +81,80 @@
     # (e.g. via a command-line flag).
     config.default_formatter = 'doc'
   end
+
+  # Check for leaky test resources
+  config.after(:all) do
+    # Exclude acceptable background threads
+    background_threads = Thread.list.reject do |t|
+      group_name = t.group.instance_variable_get(:@group_name) if t.group.instance_variable_defined?(:@group_name)
+      backtrace = t.backtrace || []
+
+      # Current thread
+      t == Thread.current ||
+        # Thread has shut down, but we caught it right as it was still alive
+        !t.alive? ||
+        # Internal JRuby thread
+        defined?(JRuby) && JRuby.reference(t).native_thread.name == 'Finalizer' ||
+        # WEBrick singleton thread for handling timeouts
+        backtrace.find { |b| b.include?('/webrick/utils.rb') } ||
+        # Rails connection reaper
+        backtrace.find { |b| b.include?('lib/active_record/connection_adapters/abstract/connection_pool.rb') } ||
+        # Ruby JetBrains debugger
+        t.class.name.include?('DebugThread') ||
+        # Categorized as a known leaky thread
+        !group_name.nil?
+    end
+
+    unless background_threads.empty?
+      info = background_threads.flat_map do |t|
+        caller = t.instance_variable_get(:@caller) || '(not recorded)'
+        [
+          "#{t} (#{t.class.name})",
+          ' == Caller ==',
+          caller,
+          ' == Backtrace ==',
+          t.backtrace,
+          "\n"
```

Comment:
If we are lucky that the thread was initialized after `DatadogThreadDebugger` was injected, we get a very nice call stack with the offending thread creator line.